dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OpenAPIGeneralIT.java

```diff
@@ -247,4 +252,66 @@ void testToolAlias() {
         assertNotNull(anonEntriesApi.getEntryByAlias("foobar"), "Should retrieve the tool by alias");
     }
+
+    /**
+     * Tests that the getPublicWorkflowVesions function excludes hidden versions correctly.
+     */
+    @Test
+    void testGetPublicWorkflowVersions() {
+        ApiClient client = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowsOpenApi = new io.dockstore.openapi.client.api.WorkflowsApi(client);
+
+        Workflow workflow = registerWorkflowWithTwoVersions();
+
+        //publish workflow
+        workflow = workflowsOpenApi.publish1(workflow.getId(), CommonTestUtilities.createOpenAPIPublishRequest(true));
+
+        List<WorkflowVersion> workflowVersions = workflowsOpenApi.getWorkflowVersions(workflow.getId(), null, null, null, null);
+        //Hide version
+        workflowVersions.get(1).setHidden(true);
+        workflowsOpenApi.updateWorkflowVersion(workflow.getId(), workflowVersions);
+
+        List<WorkflowVersion> publicWorkflowVersions = workflowsOpenApi.getPublicWorkflowVersions(workflow.getId(), null, null, null, null);
+        assertEquals(1, publicWorkflowVersions.size(), "Should exclude hidden version thus only have 1 version");
+    }
+
+    @Test
+    void testWorkflowVersionsSorting() {
+        ApiClient client = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowsOpenApi = new io.dockstore.openapi.client.api.WorkflowsApi(client);
+
+        Workflow workflow = registerWorkflowWithTwoVersions();
+        // Test sorting by name in ascending order
+        List<WorkflowVersion> workflowVersions = workflowsOpenApi.getWorkflowVersions(workflow.getId(), null, null, "name", "asc");
+        assertEquals("master", workflowVersions.get(0).getName(), "The first version should be master");
+
+        // Test sorting by name in descending order
+        workflowVersions = workflowsOpenApi.getWorkflowVersions(workflow.getId(), null, null, "name", "desc");
+        assertEquals("testCWL", workflowVersions.get(0).getName(), "The first version should be testCWL");
+    }
+
+    private Workflow registerWorkflowWithTwoVersions() {
+        ApiClient client = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowsOpenApi = new io.dockstore.openapi.client.api.WorkflowsApi(client);
+
+        // refresh all
+        workflowsOpenApi.manualRegister(SourceControl.GITHUB.name(), "DockstoreTestUser2/hello-dockstore-workflow", "/Dockstore.cwl", "",
+                DescriptorLanguage.CWL.getShortName(), "");
+
+        // refresh individual that is valid
+        io.dockstore.openapi.client.model.Workflow workflow = workflowsOpenApi
+                .getWorkflowByPath("github.com/DockstoreTestUser2/hello-dockstore-workflow", WorkflowSubClass.valueOf(BIOWORKFLOW), "");
+
+
+        // Refresh a single version
+        workflow = workflowsOpenApi.refreshVersion(workflow.getId(), "master", false);
```

Comment:
Interesting, I didn't remember that you can refresh specifically one (two) versions of a workflow. 
This is a nice way of isolating test cases